### PR TITLE
Cleanup DSpace Solr Configs for v8 & update to Solr 8.8.1

### PR DESCRIPTION
**Pull request by tdonohue**
## References
* Builds on / requires #3020 (already merged)

## Description
This PR simplifies our `solrconfig.xml` Solr core configurations by removing any settings which are "default" / unnecessary based on the reference guide at https://lucene.apache.org/solr/guide/8_7/configuring-solrconfig-xml.html 

This includes removing obviously unused `<requestHandler>` and removing all [requestHandlers which are implicit](https://lucene.apache.org/solr/guide/8_7/requesthandlers-and-searchcomponents-in-solrconfig.html#implicit-request-handlers) (defined by default)

I've also removed unnecessary configuration files/directories (including the `xslt` subdirectory for all cores)

Per feedback in #3020, I've also updated us to the latest version of Solr 8 (v8.8.1).

## Instructions for Reviewers

* Review changes
* Optionally test on a local Solr 8 (I've already tested this in Docker and found no issues so far)

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by mwoodiupui** [comment]
First look:  sensible, with a couple of inline comments.  I've been wanting to clean out ancient sample configuration for some time.  In particular, the cache warming queries made no sense for our cores and simply wasted cycles.

**Comment by mwoodiupui** [comment]
> Hmmm...looking around, I suspect the `text/plain` may mean these JSONResponseWriters are **not used** in DSpace. I found this in the Solr 8.0 docs where it notes that setting it to `text/plain` is only useful to read in your browser, not if you actually want to consume the JSON...see the comment on the example queryResponseWriter in this section: https://lucene.apache.org/solr/guide/8_0/response-writers.html#json-response-writer

SolrJ should be handling whatever kind of response document comes back, so we shouldn't need to ask for JSON anyway.

> Therefore, I think maybe we should just remove these...it doesn't seem like they could be used anywhere.

I agree.

**Comment by tdonohue** [comment]
@mwoodiupui and @abollini : This PR is now ready for review. A few minor updates:
* I've addressed all of @mwoodiupui 's prior feedback. See https://github.com/DSpace/DSpace/pull/3126/commits/575994fab1ffc3ac33efcea4eec5654f0473c11c
* I've pushed up a (very minor) update to Solr v8.8.1 (see https://github.com/DSpace/DSpace/pull/3126/commits/7b9e3d70bb437422d09583b7d9bd26cd9db87769) as requested by @abollini .  Today I've been testing the Solr v8.8.1 update in Docker and have not seen any issues.

File: dspace/solr/search/conf/solrconfig.xml [hunk 1]
```diff
@@ -23,530 +23,146 @@
...
-      </fragmenter>
+        <arr name="last-components">
+            <str>spellcheck</str>
+        </arr>
+    </requestHandler>
 
-      <fragmenter name="regex"
-                  class="solr.highlight.RegexFragmenter">
+    <!-- A request handler that returns indented JSON by default -->
+    <requestHandler name="/query" class="solr.SearchHandler">
```
Comment (mwoodiupui) [inline dspace/solr/search/conf/solrconfig.xml:126]: I don't see where we use this.

File: dspace/solr/statistics/conf/solrconfig.xml [hunk 2]
```diff
@@ -23,532 +23,123 @@
...
-      <fragmentsBuilder name="default"
-                        default="true"
-                        class="solr.highlight.ScoreOrderFragmentsBuilder">
-      </fragmentsBuilder>
+    </requestHandler>
 
-      <fragmentsBuilder name="colored"
-                        class="solr.highlight.ScoreOrderFragmentsBuilder">
+    <!-- A request handler that returns indented JSON by default -->
+    <requestHandler name="/query" class="solr.SearchHandler">
```
Comment (mwoodiupui) [inline dspace/solr/statistics/conf/solrconfig.xml:114]: I don't see where we use this.

**Review by mwoodiupui — APPROVED** [review]
MUCH less cluttered with unused examples.  I left a few inline comments but this should work regardless.

**Comment by tdonohue** [comment]
@mwoodiupui : Good point on the JSON request handlers -- they don't appear to be used.  I removed them, rebuilt & tested REST API & OAI, and I didn't hit any issues.  So, just pushed up that small fix too: https://github.com/DSpace/DSpace/pull/3126/commits/0cd1e1b63612e21662bce2fe286e8e73c9395d2f

**Review by abollini — APPROVED** [review]
tested all looks fine, merging right now

**Pull request merged**